### PR TITLE
Pass eventtype owner to data plane

### PR DESCRIPTION
**Pull request by Cali0707**
Part of #3181

We need the eventtype Reference/Owner information in the dataplane so that we can auto create event types in #3181.

## Proposed Changes

- Pass the event type ownership info through the data plane contract

### CODE REVIEW

**Comment by Cali0707** [comment]
I think the upgrade test failures might be real failures, I will look into it

File: control-plane/pkg/reconciler/broker/broker.go [hunk 1]
```diff
@@ -608,12 +608,27 @@ func rebuildCMFromStatusAnnotations(br *eventing.Broker) *corev1.ConfigMap {
...
 		Ingress: &contract.Ingress{
 			Path:                       receiver.PathFromObject(broker),
 			EnableAutoCreateEventTypes: feature.FromContext(ctx).IsEnabled(feature.EvenTypeAutoCreate),
+			EventTypeOwnerReference: &contract.Reference{
+				Uuid:      string(broker.UID),
+				Name:      broker.Name,
+				Namespace: broker.Namespace,
+				Kind:      brokerKind,
+				Version:   brokerAPIVersion,
+			},
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/broker/broker.go:631]: Why not including this in `reference` field?

Comment (pierDipi) [inline control-plane/pkg/reconciler/broker/broker.go:631]: I would simplify this to something like:
```suggestion
	resource := &contract.Resource{
		Uid:    string(broker.UID),
		Topics: []string{topic},
		Ingress: &contract.Ingress{
			Path:                       receiver.PathFromObject(broker),
			EnableAutoCreateEventTypes: feature.FromContext(ctx).IsEnabled(feature.EvenTypeAutoCreate),
			EventTypeOwnerReference: &contract.Reference{
				Uuid:      string(broker.UID),
				Name:      broker.Name,
				Namespace: broker.Namespace,
				Kind:      "Broker",
				Version:   eventing.SchemeGroupVersion.String(),
			},
```

Comment (Cali0707) [reply control-plane/pkg/reconciler/broker/broker.go:631]: @pierDipi is there a different reference field I could be including them on?

Comment (pierDipi) [reply control-plane/pkg/reconciler/broker/broker.go]: Yes, it's called `Reference` under `Resource`

https://github.com/knative-extensions/eventing-kafka-broker/blob/0242a8795c39a50266570f3cf3211bbabbd209b5/proto/contract.proto#L333-L339

**Comment by Cali0707** [comment]
@Leo6Leo I think the test failure is an actual issue, not a flaky failure here. Not sure what is causing the issue though...

File: control-plane/pkg/reconciler/testing/objects_channel.go [hunk 2]
```diff
@@ -259,6 +259,8 @@ func ChannelReference() *contract.Reference {
 		Uuid:      ChannelUUID,
 		Namespace: ChannelNamespace,
 		Name:      ChannelName,
+		Kind:      "KafkaChannel",
+		Version:   messagingv1beta1.SchemeGroupVersion.String(),
```
Comment (pierDipi) [inline control-plane/pkg/reconciler/testing/objects_channel.go:263]: I think `Version` here is meant to be `ResourceVersion` as opposed to `GroupVersion`

Comment (Cali0707) [reply control-plane/pkg/reconciler/testing/objects_channel.go:263]: Oh okay, I need the group version so I will add a new field for that then. Thanks!

**Comment by Cali0707** [comment]
Now that changes to the contract are backwards compatible, the tests are passing on this PR.

/cc @pierDipi @Leo6Leo

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**